assembly/src/tokens/stream.rs

```diff
@@ -21,15 +28,34 @@ impl<'a> TokenStream<'a> {
         if source.is_empty() {
             return Err(AssemblyError::empty_source());
         }
+        let mut doc_comments = BTreeMap::new();
+
+        let mut tokens = Vec::new();
+        let mut comment = Comment(String::new());
 
-        let tokens = source
-            .lines()
-            // Tokenize and remove comments
-            .flat_map(|line| {
-                line.split_whitespace()
+        let lines = source.lines();
+        for line in lines {
+            let line = line.trim();
+            if line.starts_with(DOC_COMMENT_PREFIX) {
+                comment.append_line(line);
+            } else if line.starts_with(FUNC_PREFIX) {
+                doc_comments.insert(tokens.len(), comment.get_comment());
+                comment.clear();
+                let mut line_tokens = line
+                    .split_whitespace()
+                    .take_while(|&token| !token.starts_with('#'))
+                    .collect::<Vec<_>>();
+                tokens.append(&mut line_tokens);
+            } else if line.is_empty() {
+                comment.clear();
```

Comment:
I am not sure we should do this here. If there are breaks in the comments, we should either: show an error (or a warning) or assume that it is the same comment.

Showing a warning would probably be best, but we don't have this mechanism now. So, maybe we should return an error? (something like `MalformedDocComment`).